nominal/core/connection.py

```diff
@@ -141,12 +151,59 @@ def get_channel(self, name: str, tags: dict[str, str] | None = None) -> Channel:
         series = self._clients.logical_series.get_logical_series(self._clients.auth_header, resolved_series.rid)
         return Channel._from_conjure_logicalseries_api(self._clients, series)
 
-    def get_nominal_write_stream(self, batch_size: int = 50_000, max_wait_sec: int = 1) -> WriteStream:
-        """get_nominal_write_stream is deprecated and will be removed in a future version,
-        use get_write_stream instead.
+    def archive(self) -> None:
+        """Archive this connection.
+        Archived connections are not deleted, but are hidden from the UI.
         """
+        self._clients.connection.archive_connection(self._clients.auth_header, self.rid)
+
+    def unarchive(self) -> None:
+        """Unarchive this connection, making it visible in the UI."""
+        self._clients.connection.unarchive_connection(self._clients.auth_header, self.rid)
+
+
+@dataclass(frozen=True)
+class StreamingConnection(Connection):
+    nominal_data_source_rid: str
+
+    def get_multiprocessed_write_stream(
+        self,
+        max_batch_size: int = 30000,
+        max_wait: timedelta = timedelta(seconds=1),
+        max_queue_size: int = 0,
+        max_workers: int = 10,
+    ) -> MultiProcessedWriteStream:
+        """Stream to write non-blocking messages to a datasource.
+
+        Args:
+        ----
+            max_batch_size (int): How big the batch can get before writing to Nominal. Default 50,000
```

Comment:
Don't duplicate types and defaults in the docstring - the signature already covers it and it gives too many opportunities for inconsistencies (like this here).